controllers/subscription_controller.go

```diff
@@ -85,7 +91,17 @@ func (r *SubscriptionReconciler) Reconcile(ctx context.Context, _ ctrl.Request)
 		return ctrl.Result{}, err
 	}
 
-	if err := reconcileCsvWebhook(ctx, r.Client, logger, r.OperatorNamespace); err != nil {
+	targetNamespaces := getTargetNamespaces(olmPkgRecords)
+
+	if err := reconcileCsvWebhook(ctx, r.Client, logger, r.OperatorNamespace, targetNamespaces); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := ensureNamespace(ctx, r.Client, logger, olmPkgRecords); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	if err := ensureOperatorGroup(ctx, r.Client, logger, olmPkgRecords); err != nil {
```

Comment:
why are these funcs rather than being methods on the reconciler?

Reply:
Just like that, We can convert them.

Reply:
Done